### ISSUE TITLE
Assert descriptions

### ISSUE THREAD

**Issue by Timoses** [jmdavis/dxml#14]
https://github.com/jmdavis/dxml/blob/14b9d0bb245049c730d2d68641819a0c91a64a6c/source/dxml/dom.d#L685

Such asserts could really use some description, in this case e.g.:
```
assert(_type == EntityType.elementStart, "Only start elements have children"); 
```

for improved debugging when using the library.

**Edit:**
In this particular case I'm not even sure if it's necessary to force the assert. Returning an empty children array might be more user-friendly.
```
if (node.children.length)
```
vs.
```
if (node.type == EntityType.elementStart && node.children)
```

**Comment by jmdavis** [comment]
Most of those top-level assertions have messages have already. I thing that you may have found the only one that doesn't.

**Comment by jmdavis** [comment]
Fixed in commit 4654066e93e74a9b399358d8c9938146bd557a2d
